xbrl_parser/instance.py

```diff
@@ -566,3 +566,23 @@ def _parse_unit_elements(unit_elements: List[ET.Element]) -> dict:
                               divide.find('xbrli:unitDenominator/xbrli:measure', NAME_SPACES).text.strip())
         unit_dict[unit_id] = unit
     return unit_dict
+
+
+class XbrlParser:
+    """
+    XbrlParser to make interaction easier.
+
+    """
+
+    def __init__(self, cache: HttpCache):
+        self.cache = cache
+
+    def parse_xbrl(self, url: str) -> XbrlInstance:
+        instance_path: str = self.cache.cache_file(url)
+
+        if url.split('.')[-1] == 'xml':
+            return parse_xbrl(instance_path, self.cache, url)
+        return parse_ixbrl(instance_path, self.cache, url)
```

Comment:
The only change i would make is to rename this function to something like `parse_instance`. 
Just looking at the name, it is not clear if this function parses XBRL taxonomies, linkbases or instance documents. 
I can rename it in my next commit :)

Reply:
That makes sense, the naming you suggest is a lot clearer. By the way, you can add commits on a pull request before approval. That way is easier to track changes. Feel free to add commits to any of my pull requests in the future :D instead of adding the commit outside of the pull request.

Reply:
Yes, sorry i only thought about suggestion after i already merged the pull request.
I wanted to rename it myself in my next commit, but you were faster with the new pull request 😄.